src/nvim/terminal.c

```diff
@@ -907,6 +928,16 @@ static void on_refresh(Event event)
       }
       continue;
     }
+    FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
+      if (wp->w_buffer == term->buf) {
+        if ( wp->w_botline > wp->w_buffer->b_ml.ml_line_count ) {
+          // if last line is visible, follow the terminal
+          wp->following_terminal = true;
+        } else {
+          wp->following_terminal = false;
+        }
```

Comment:
I see. Well, I'd be appeased if the terminal only scrolled when the cursor sat on the last line, and preferably only if at the end, so I could split the window and just move one line up to disable scrolling. If the terminal only has one line, I could just move to the left.

Worst case, we could make it a buffer-local option whether to scroll unconditionally, but that doesn't seem like a nice workflow (`:set noscroll | sp  | *do comamnd*`).